pkg/querier/querier_test.go

```diff
@@ -320,86 +316,6 @@ func TestQuerier_QueryableReturnsChunksOutsideQueriedRange(t *testing.T) {
 	}, m[0].Floats)
 }
 
-// TestBatchMergeChunks is a regression test to catch one particular case
-// when the Batch merger iterator was corrupting memory by not copying
-// Batches by value because the Batch itself was not possible to copy
-// by value.
-func TestBatchMergeChunks(t *testing.T) {
-	var (
-		logger     = log.NewNopLogger()
-		queryStart = mustParseTime("2021-11-01T06:00:00Z")
-		queryEnd   = mustParseTime("2021-11-01T06:01:00Z")
-		queryStep  = time.Second
-	)
-
-	var cfg Config
-	flagext.DefaultValues(&cfg)
-	cfg.BatchIterators = true // Always use the Batch iterator - regression test
```

Comment:
given this was testing _with_ batch iterators, then should we also keep that test (assuming that batch iterators are always on from now on)?

Reply:
I think we want to restore it. It was introduced earlier this year: https://github.com/grafana/mimir/pull/4376

Reply:
restored in https://github.com/grafana/mimir/pull/6814